### PR TITLE
Test Case of IDU finalized with complete edge case files

### PR DESCRIPTION
**Pull request by Shhhhhubh**
Addresses
- add test cases for IDU

## Changes

* Detailed list or prose of changes
* Breaking changes
* Changes to configurations

## This PR doesn't introduce any:

- [ ] temporary files, auto-generated files or secret keys
- [ ] n+1 queries
- [ ] linter issues
- [ ] `print`
- [ ] typos
- [ ] unwanted comments

## This PR contains valid:

- [ ] tests

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -135,3 +136,5 @@ dmypy.json
 assets/
 media/
 *.gpkg
+
+apps/etl/Dataset
```
Comment (tnagorra) [inline .gitignore]:
```suggestion
apps/etl/dataset
```

File: Dockerfile [hunk 2]
```diff
@@ -22,6 +22,7 @@ RUN --mount=type=cache,target=/root/.cache/uv \
     apt-get update -y \
     && apt-get install -y --no-install-recommends \
         # Build required packages
+        build-essential libgdal-dev \
```
Comment (tnagorra) [inline Dockerfile]: Why are we adding this just for the tests?

File: libs/pystac-monty
Comment (tnagorra) [inline libs/pystac-monty:1]: Check if this is the latest pystac-monty. This should not be updated.

File: pyproject.toml [hunk 3]
```diff
@@ -35,6 +35,14 @@ dependencies = [
...
     "termcolor",
     "ijson>=3.4.0",
+    # TODO: Move to dev dependencies
+    "pytest",
+    "requests-mock",
+    "pytest-django",
+    "pytest-icdiff",
+    "pytest-ordering",
+    "pytest-profiling",
+    "json5",
```
Comment (tnagorra) [inline pyproject.toml]: Let's move this to dev dependencies. Also, are we using all of these libraries?

File: apps/etl/tests/Dataset/IDMC-IDU/fixed_output_idu.json
Comment (tnagorra) [inline apps/etl/tests/Dataset/IDMC-IDU/fixed_output_idu.json:1]: Let's rename these to `dataset/idmc_idu/`

@thenav56 Should we use kebab case or snake case for the filenames?

File: apps/etl/tests/sources/test_idu.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    if isinstance(obj, dict):
+        for key in list(obj.keys()):
+            if key in keys_to_ignore:
+                obj.pop(key)
+            else:
+                remove_ignored_keys(obj[key], keys_to_ignore)
+    elif isinstance(obj, list):
+        for item in obj:
+            remove_ignored_keys(item, keys_to_ignore)
+    return obj
```
Comment (tnagorra) [inline apps/etl/tests/sources/test_idu.py]: Let's move this to utils.

File: apps/etl/tests/sources/test_idu.py [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        ext_and_transform_idu_latest_data()
+
+    # Serialize queryset to JSON string
+    latest_data = PyStacLoadData.objects.all()
+    latest_data_json = serialize("json", latest_data)
+
+    # Use a temporary file to store the output
+    with tempfile.NamedTemporaryFile(mode="w+", suffix=".json", delete=False, encoding="utf-8") as temp_file:
+        temp_file.write(latest_data_json)
+        output_path = Path(temp_file.name)
```
Comment (tnagorra) [inline apps/etl/tests/sources/test_idu.py]: We don't need to write this to a file for comparison
cc: @thenav56

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: apps/etl/tests/Dataset/IDMC-IDU/fixed_output_idu.json
Comment (thenav56) [reply apps/etl/tests/Dataset/IDMC-IDU/fixed_output_idu.json:1]: I like kebab case for non-python files/directories.
But if it's confusing, then do snake-case for all
